src/compose.ts

```diff
@@ -65,9 +66,28 @@ services:
  * does not have the expected structure and discrepancies can't be resolved,
  * validation errors are thrown. The input composition is mutated in-place.
  *
- * @param c The input composition as a plain JS object
+ * @param o The input composition as a plain JS object
+ * @param fileResolverCb Callback to access filePaths
+ * 	Has to validate that no symbolic links are used outside project folder
```

Comment:
If I was using this module, I'd want to see a short paragraph documenting what passing a callback means, how it's used and what it enables. That is, a brief description of the feature from the perspective of the user.

Reply:
I've added more context for the usage of the callback